lcls_tools/common/measurements/screen_profile.py

```diff
@@ -82,29 +79,36 @@ def measure(self, n_shots: int = 1) -> dict:
         shot number
         """
         images = []
-        while len(images) < n_shots:
-            images.append(self.device.image)
+        while len(images) < self.n_shots:
+            images.append(self.beam_profile_device.image)
             # TODO: need to add a wait statement in here for images to update
 
         processed_images = [
             self.image_processor.auto_process(image) for image in images
         ]
 
         if self.fit_profile:
-            rms_sizes = []
-            centroids = []
-            total_intensities = []
+            rms_sizes_all = []
+            centroids_all = []
+            total_intensities_all = []
             for image in processed_images:
                 fit_result = self.beam_fit.fit_image(image)
-                rms_sizes.append(fit_result.rms_size)
-                centroids.append(fit_result.centroid)
-                total_intensities.append(fit_result.total_intensity)
+                rms_sizes_all.append(
+                    np.array(fit_result.rms_size) * self.beam_profile_device.resolution
+                )
+                centroids_all.append(
+                    np.array(fit_result.centroid) * self.beam_profile_device.resolution
+                )
+                total_intensities_all.append(fit_result.total_intensity)
+            rms_sizes = np.mean(rms_sizes_all, axis=0)
```

Comment:
we should do this averaging outside the screen profile class to enable flexibility. For the emittance class you can check the beamsize measurement class type to see if you have to do the averaging

Reply:
thats a good point, maybe we should return the mean values as `mean_rms_sizes` and use that in the emittance class